src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -1112,14 +1126,16 @@ endif
         jmp     RetrySyncBlock
 
     Block:
-        ; In the Block case we've trashed RCX, restore it
+        ; In the Block case we've trashed RCX and RDX restore them
         mov     rcx, [rsp + 8h]
+        mov     rdx, [rsp + 10h]
```

Comment:
There is "mov     rdx, [rsp + 10h]" instruction right before "jmp     JITutil_MonTryEnter" that will take care of restoring rdx. This change should not be necessary.